src/aiida_quantumespresso/workflows/xps.py

```diff
@@ -366,7 +376,7 @@ def get_treatment_filepath(cls):
     @classmethod
     def get_builder_from_protocol(
         cls, code, structure, pseudos, core_hole_treatments=None, protocol=None,
-        overrides=None, elements_list=None, options=None,
+        overrides=None, elements_list=None, atoms_list=None, options=None,
         structure_preparation_settings=None, **kwargs
     ):
```

Comment:
Adding at least the `correction_energies` in the form of a standard Python `dict` object would probably be enough - since they are required for a calculation where `calc_binding_energy = True`, the function could just set `calc_binding_energy = orm.Bool(True)` if `correction_energies` is defined at all. If they aren't defined, then the function should logically set `calc_binding_energy = orm.Bool(False)`

```
def get_builder_from_protocol(
    ...        
    structure_preparation_settings=None, correction_energies=None, **kwargs
    ):
```
```
if correction_energies:
    builder.correction_energies = orm.Dict(correction_energies)
    builder.calc_binding_energies = orm.Bool(True)
else:
    builder.calc_binding_energies = orm.Bool(False)
```